### PR TITLE
Runable Lexer Update

### PR DESCRIPTION
**Pull request by berkenvr**
Lexer can now be tested. It includes basic tokens and keywords. All functions work as intended. It needs simple improvements in some areas (you can check TODOs). Please let me know if you notice anything, thanks!

### CODE REVIEW

File: src/tile.c [hunk 1]
```diff
@@ -1,10 +1,40 @@
 #include <stdio.h>
+#include "../include/tile_lexer.h"
```
Comment (yasinxdxd) [inline src/tile.c]: just use <tile_lexer>

File: src/tile_lexer.c [hunk 2]
```diff
@@ -0,0 +1,219 @@
+#define ARENA_IMPLEMENTATION
+#include "../include/tile_lexer.h"
```
Comment (yasinxdxd) [inline src/tile_lexer.c]: just use <tile_lexer>

File: src/tile_lexer.c [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+#include <stdlib.h>
+#include <string.h>
+#include <ctype.h>
+#include <../include/common/arena.h>
```
Comment (yasinxdxd) [inline src/tile_lexer.c]: <...>

File: src/tile_lexer.c [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+
+    { NULL, TOKEN_NONE },
+};
+
+static const tile_lextoken_t keywords[] = {
+    { "true", TOKEN_TRUE },
+    { "false", TOKEN_FALSE }, // TODO: true and false can also uppercase like TRUE, FALSE, True, False. Find a solution for that.
+
+    { "if", TOKEN_IF },
+    { "else if", TOKEN_ELSEIF }, // TODO: 'else if' or 'elseif' which one?
```
Comment (yasinxdxd) [inline src/tile_lexer.c]: there is no such a thing, "else if" is not a single token. look at my comment at issue #3

File: include/tile_token.h [hunk 5]
```diff
@@ -0,0 +1,53 @@
+#ifndef TOKEN_H
+#define TOKEN_H
+
+#include <stdio.h>
+
+typedef enum {
+    TOKEN_NONE,
+
+    // Literals
```
Comment (yasinxdxd) [inline include/tile_token.h:9]: it would be good to add "_LITERAL" suffix at the end

File: include/tile_token.h [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    TOKEN_TRUE,
+    TOKEN_FALSE,
+    TOKEN_STRING,
+    TOKEN_INT,
+    TOKEN_FLOAT,
+    TOKEN_ID,
+
+    // Keywords
+    TOKEN_IF,
+    TOKEN_ELSEIF,
```
Comment (yasinxdxd) [inline include/tile_token.h]: it is not a thing, look at issue #3

File: src/tile_token.c [hunk 7]
```diff
@@ -0,0 +1,10 @@
+#include "../include/tile_token.h"
```
Comment (yasinxdxd) [inline src/tile_token.c]: just use <tile_token.h>

**Review by yasinxdxd — CHANGES_REQUESTED** [review]

File: include/tile_token.h (hunk 5 above)
Comment (yasinxdxd) [reply include/tile_token.h:9]: "_LITERAL" suffix after all literals

File: src/tile_lexer.c (hunk 3 above)
Comment (yasinxdxd) [reply src/tile_lexer.c]: <common/arena.h>

**Review by berkenvr — COMMENTED** [review]
I checked all comments and applied them. The lexer is working correctly.

**Review by yasinxdxd — APPROVED** [review]
LGTM!

**Pull request merged**